### PR TITLE
feat(argo-cd): move server.additionalApplications and server.additionalProjects to a new chart

### PR DESCRIPTION
**Pull request by yu-croco**
In order to decide how to manage [argo-cd/crds](https://github.com/argoproj/argo-helm/tree/main/charts/argo-cd/crds), I removed values from argo-cd.
*After merging https://github.com/argoproj/argo-helm/pull/1356, I will merge this PR.

- other reference: https://github.com/argoproj/argo-helm/pull/1342#pullrequestreview-1019481838

---

Signed-off-by: yu-croco <yuki.kita22@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/README.md
Comment (yu-croco) [inline charts/argo-cd/README.md:260]: They are moved to argocd-apps, https://github.com/argoproj/argo-helm/pull/1356  🙋

File: charts/argo-cd/README.md
Comment (mkilchhofer) [inline charts/argo-cd/README.md:135]: We could also mention another method to do a `helm upgrade` of the existing chart version and setting 


```yaml
server:
  additionalApplications:
    - name: guestbook
      namespace: argocd
      additionalLabels: {}
      additionalAnnotations:
        "helm.sh/resource-policy": keep # <-- see this
      finalizers:
      - resources-finalizer.argocd.argoproj.io
      project: guestbook
      source:
        repoURL: https://github.com/argoproj/argocd-example-apps.git
        targetRevision: HEAD
        path: guestbook
        directory:
          recurse: true
      destination:
        server: https://kubernetes.default.svc
        namespace: guestbook
      syncPolicy:
        automated:
          prune: false
          selfHeal: false
      ignoreDifferences:
      - group: apps
        kind: Deployment
        jsonPointers:
        - /spec/replicas
      info:
      - name: url
        value: https://argoproj.github.io/
```

File: charts/argo-cd/Chart.yaml
Comment (mkilchhofer) [inline charts/argo-cd/Chart.yaml:24]:
```suggestion
    - "[Removed]: Parameters server.additionalApplications and server.additionalProjects are moved to a separate chart - argocd-apps"
```

File: charts/argo-cd/README.md
Comment (mkilchhofer) [inline charts/argo-cd/README.md:99]: Add something like

```suggestion

This release **removes the parameters** `server.additionalApplications` and `server.additionalProjects`. Please carefully read the following section if you are using these parameters!
```

**Comment by yu-croco** [comment]
Thank you for reviewing, @mkilchhofer !
I fixed on https://github.com/argoproj/argo-helm/pull/1357/commits/7ddf911bb9a383c64c91d39408436adb0fc52f46 . 🙋

**Review by mkilchhofer — APPROVED** [review]

**Comment by yu-croco** [comment]
I will merge this PR after merging https://github.com/argoproj/argo-helm/pull/1356 🙋

**Review by mbevc1 — APPROVED** [review]

**Comment by mkilchhofer** [comment]
Implemented in PR https://github.com/argoproj/argo-helm/pull/1417

**Pull request closed without merging**
